Mathlib/LinearAlgebra/TensorProduct/Basis.lean

```diff
@@ -66,5 +66,44 @@ lemma Basis.baseChange_apply (b : Basis ι R M) (i) :
     b.baseChange S i = 1 ⊗ₜ b i := by
   simp [Basis.baseChange, Basis.tensorProduct]
 
+section
+
+variable (ℬ : Basis ι R M) (𝒞 : Basis κ R N) (x : M ⊗[R] N)
+
+/-- Elements in `M ⊗ N` can be represented by sum of elements in `M` tensor elements of basis of
+`N`. -/
+lemma TensorProduct.eq_repr_basis_right :
+    ∃ b : κ →₀ M, b.sum (fun i m => m ⊗ₜ 𝒞 i) = x := by
+  classical
+  induction x using TensorProduct.induction_on with
+  | zero => exact ⟨0, by simp⟩
+  | tmul m n =>
+    use 𝒞.repr n |>.mapRange (fun (r : R) => r • m) (by simp)
```

Comment:
```suggestion
    use (𝒞.repr n).mapRange (fun (r : R) ↦ r • m) (by simp)
```
The `.|>` trick is nice but in my opinion makes things harder to read.

Similarly elsewhere.

Reply:
Changed thanks